### PR TITLE
Feature/crosslisting v2 search courses

### PR DESCRIPTION
**Pull request by sapnamysore**
This PR has changes for  [3680](https://jira.huit.harvard.edu/browse/TLT-3680)

Made thee primary and secondary inputs search boxes with  a type ahead capability. 

@cmurtaugh @Chris-Thornton-Harvard

### CODE REVIEW

File: cross_list_courses/views.py [hunk 1]
```diff
@@ -99,3 +105,46 @@ def create_new_pair(request):
...
+@lti_role_required(const.ADMINISTRATOR)
+@lti_permission_required(settings.PERMISSION_XLIST_TOOL)
+@require_http_methods(['GET'])
+def get_ci_data(request):
+    today = datetime.now()
+
+    if request.is_ajax():
+        q = request.GET.get('term', '')
+        logger.debug(q)
+        print q
```
Comment (Chris-Thornton-Harvard) [inline cross_list_courses/views.py]: Removal of print statement

File: cross_list_courses/views.py [hunk 2]
```diff
@@ -99,3 +105,46 @@ def create_new_pair(request):
...
+        courses = CourseInstance.objects.filter(Q(term__end_date__gte=today - timedelta(days=120))).filter(
+            Q(course_instance_id__icontains=q)|
+            Q(title__icontains=q)|
+            Q(sub_title__icontains=q) |
+            Q(short_title__icontains=q)).filter(Q(cs_class_type='E') |
+                                                Q(cs_class_type__isnull=True)).select_related('term')[:10]
```
Comment (Chris-Thornton-Harvard) [inline cross_list_courses/views.py]: Should the slice be included?

File: cross_list_courses/templates/add_new.html [hunk 3]
```diff
@@ -107,10 +107,43 @@ <h3>
             "bInfo" : false,
             "aoColumnDefs": [ { "orderable": false, "targets": 2 } ],
             "oLanguage": {
-              "sSearch": "Search"
+              "sSearch": "Filter"
             }
         });
     });
+
+    $(document).ready(function(){
```
Comment (Chris-Thornton-Harvard) [inline cross_list_courses/templates/add_new.html]: Should this function be removed?

File: cross_list_courses/views.py [hunk 4]
```diff
@@ -99,3 +105,46 @@ def create_new_pair(request):
...
+        # logger.debug(courses.query)
```
Comment (Chris-Thornton-Harvard) [inline cross_list_courses/views.py]: Commented out logging statement removal

**Comment by Chris-Thornton-Harvard** [comment]
I get the following error in my web console when I enter 2 characters in either field, "VM14:1 GET https://canvas-account-admin-tools.qa.tlt.harvard.edu/cross_list_courses/get_ci_data?term=sa&resource_link_id=ca122ea3b565b2faf8882c3e1da0ddf02973672a 500 (Internal Server Error)"

File: cross_list_courses/templates/add_new.html [hunk 5]
```diff
@@ -107,10 +107,43 @@ <h3>
...
+            var location_input=$('input[id="primary-course"]');
+            location_input.autocomplete({
+                source: "get_ci_data",
+              minLength: 2
+            });
+          } );
+
+
+    $(document).ready(function(){
+            var location_input=$('input[id="secondary-course"]');
```
Comment (Chris-Thornton-Harvard) [inline cross_list_courses/templates/add_new.html]: Should these vars be named uniquely? Not sure how JS handles this type of stuff during the .ready() call

File: cross_list_courses/views.py (hunk 2 above)
Comment (sapnamysore) [reply cross_list_courses/views.py]: yes, 10 seems to be the number of entries we want to display in the dropdown. we don't want to display all. as they enter more characters, the results will filter down

**Comment by sapnamysore** [comment]
@Chris-Thornton-Harvard : I redeployed changes with some cleanup. also the random console error is no longer happening - handled it by encoding thee title/subtitle while building the results.

**Comment by Chris-Thornton-Harvard** [comment]
👍

**Pull request merged**